libraries/anvill_passes/src/InstructionFolderPass.cpp

```diff
@@ -133,13 +137,20 @@ const std::unordered_map<std::uint32_t, PHINodeInstructionFolder> kPHINodeFolder
   { llvm::Instruction::SIToFP, InstructionFolderPass::FoldPHINodeWithCastInst },
   { llvm::Instruction::FPTrunc, InstructionFolderPass::FoldPHINodeWithCastInst },
   { llvm::Instruction::FPExt, InstructionFolderPass::FoldPHINodeWithCastInst },
+
+
+  // NOTE: The folding of phi nodes with PtrToInt, IntToPtr, and BitCast is disabled
+  //       in the function since bouncing these into each other can cause folding
+  //       to enter into infinite loop.
+  //
   { llvm::Instruction::PtrToInt, InstructionFolderPass::FoldPHINodeWithCastInst },
```

Comment:
Can you disable via commenting our here, rather than checking with an `if` statement below?